xbmc/utils/URIUtils.cpp

```diff
@@ -466,7 +466,9 @@ std::string URIUtils::ChangeBasePath(const std::string &fromPath, const std::str
 CURL URIUtils::SubstitutePath(const CURL& url, bool reverse /* = false */)
 {
   const std::string pathToUrl = url.Get();
-  return CURL(SubstitutePath(pathToUrl, reverse));
+  CURL url_tmp(CURL(SubstitutePath(pathToUrl, reverse)));
+  url_tmp.SetCustomRequest(url.GetCustomRequest());
+  return url_tmp;
```

Comment:
The existing concept at this place is soooooo ugly:

- build an URL String from all members (get()), this is [Protocol]/URL/....|Header|options
- Then build a copy of this (wich means exploding all the things wich was concatted a line above) and substituting path.

Never ever in my life seen such a waste of resources.

But: I don't see any other place. 

Cold you pls. so nice and let me know wich place is better?
Concatting to the string (get) is imo WRONG!